### DISCUSSION TITLE
Automatic Update

### DISCUSSION THREAD

**Discussion by jmackxiii** [FlareSolverr/FlareSolverr#302] [Ideas]
I was thinking of a way to maybe check to see what version of FlareSolverr is live on a system.. perhaps by checking output from looking at the "webpage" display.. Example Below.. and compare that agaisnt the current published version here.. I have a simialr script for Ombi and it works great (can post here if needed), but my powershell isnt great, but I will play around with this, but curious if anyone had anything already for a windows based host. 

{"msg":"FlareSolverr is ready!","version":"v2.2.1","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0"}

**Comment by jmackxiii** [comment]
So after spending some time on this and looking at my Ombi Script I have I was able to reverse engineer that script to accommodate this function. Feel free to modify this to meet your needs and improve upon this if you can.. I have this scheduled as a Scheduled Task running with the highest privileges to be able to modify the service status. Use this at your own risk and I can only offer minimal support if this does not work for you. 

`#Checked the Current Version of FlareSolverr by looking at the Output of the http://localhost:8191/
function Get-CurrentDetails
{
try {
$result = Invoke-RestMethod -Uri http://localhost:8191/ -Headers $HeaderValues -UseBasicParsing
return $result
} catch {
$result = $_.Exception.Response.GetResponseStream()
$reader = New-Object System.IO.StreamReader($result)
$reader.BaseStream.Position = 0
$reader.DiscardBufferedData()
return $result
}
return $result
}

#Checks the latest version of FlareSolverr offered up on the GitHub Repo
function Get-NewVersion
{
$tag = (Invoke-WebRequest "https://api.github.com/repos/FlareSolverr/FlareSolverr/releases" | ConvertFrom-Json)[0].tag_name
return $tag
}

#Variables and Update to match your setup
$ServiceName = "FlareSolverr"
$ServiceStatus = Get-Service $ServiceName
$zip = "flaresolverr-v$FlareSolverrUpdate-windows-x64.zip"
$file = "$tempfolder\$zip"
$workingdir = "C:\"
$tempfolder = "C:\FlareSolverrTemp\"
$FlareSolverrCurrent = Get-CurrentDetails -HeaderValues $headers
$FlareSolverrUpdate = Get-NewVersion

#Compares the Running Version with the latest released version
if ($FlareSolverrCurrent.version -eq $FlareSolverrUpdate)
{
        write-host No Update Needed
}
else
{
        write-host Update Needed
        #Downloads the current Windows x64 Zip from the Repo
        $download = "https://github.com/FlareSolverr/FlareSolverr/releases/download/$FlareSolverrUpdate/flaresolverr-$FlareSolverrUpdate-windows-x64.zip"
        Invoke-WebRequest $download -Out $file
        #Stops the Serice
        Stop-Service $ServiceName 
        #Deletes the old working files
        Remove-Item $workingdir\flaresolverr\ -Recurse -Force -ErrorAction SilentlyContinue
        #Unzips the new download to the temp directory
        Expand-Archive -LiteralPath $file -DestinationPath $tempfolder 
        #Moves the newly unzipped files to the working directory
        Move-Item -Path $tempfolder\flaresolverr\ -Destination $workingdir -Force
        #File Cleanup
        Remove-Item $file -Force -ErrorAction SilentlyContinue
        Remove-Item $tempfolder\flaresolverr\ -Recurse -Force -ErrorAction SilentlyContinue
        #Starts the Service
        Start-Service $ServiceName 
    }

`

**Comment by ngosang** [comment]
There is an open issue to implement this feature.
https://github.com/FlareSolverr/FlareSolverr/issues/338

**Comment by sunnyd24** [comment]
> So after spending some time on this and looking at my Ombi Script I have I was able to reverse engineer that script to accommodate this function. Feel free to modify this to meet your needs and improve upon this if you can.. I have this scheduled as a Scheduled Task running with the highest privileges to be able to modify the service status. Use this at your own risk and I can only offer minimal support if this does not work for you.

@jmackxiii Many thanks for this script, it works with modifications to 2 lines which allow for changes made to the GitHub repository since v3 was released; changes are below for others using the script:

`if ("v" + $FlareSolverrCurrent.version -eq $FlareSolverrUpdate)`

`$download = "https://github.com/FlareSolverr/FlareSolverr/releases/download/$FlareSolverrUpdate/flaresolverr_windows_x64.zip"`

_Remember to disable Windows Defender SmartScreen due to unsigned app._

@ngosang Hopefully this feature will be implemented on the windows binary soon!
